### PR TITLE
GH-4362 Add syncing status

### PR DESCRIPTION
**Pull request by i-am-chitti**
This PR adds a `syncing` status on frontend.

Issue
- #4362

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsSynchronizationStatus.tsx [hunk 1]
```diff
@@ -1,27 +1,46 @@
+import { MessageChannelSyncStatus } from '@/accounts/types/MessageChannel';
 import { Status } from '@/ui/display/status/components/Status';
 
 export type SettingsAccountsSynchronizationStatusProps = {
-  syncStatus: 'synced' | 'failed' | 'notSynced';
+  syncStatus: MessageChannelSyncStatus;
+};
+
+const getColor = (syncStatus?: MessageChannelSyncStatus | null) => {
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsSynchronizationStatus.tsx:8]: I've just noticed we defined colors and label in message-channel object metadata already. Maybe we want to get theme from here?

```
  let objectMetadataItem = useRecoilValue(
    objectMetadataItemFamilySelector({
      objectName: CoreObjectNameSingular.MessageChannel
      objectNameType: 'singular',
    }),
  );
  ```
  Then get object from field and...

```
  const selectedOptions = field.metadata.options.filter(
    (option) => fieldValues?.includes(option.value),
  );
 ```
 
 What do you think @lucasbordeau? Would you recommend this approach to access the option values?

**Review by FelixMalfait — COMMENTED** [review]
Great! Just a small comment to avoid duplicating source of truth for colors

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsSynchronizationStatus.tsx (hunk 1 above)
Comment (charlesBochet) [reply packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsSynchronizationStatus.tsx:8]: Yes, we should use the options here. you can get the objectMetadataItem using useObjectMetadataItem hook directly

Comment (i-am-chitti) [reply packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsSynchronizationStatus.tsx]: @FelixMalfait,
I see options are present here - 
image

**Comment by i-am-chitti** [comment]
@FelixMalfait, @charlesBochet, Addressed the feedback. Have a re-review.
One thing to note is that, on my local, I'm always getting syncStatus as null - 
image
when I logged https://github.com/twentyhq/twenty/blob/main/packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsMessageChannelsListCard.tsx#L53 with the new changes as in the PR. 

How can I make this in other status for testing?

File: packages/twenty-front/src/modules/object-metadata/hooks/useGetSyncStatusMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { isUndefinedOrNull } from '~/utils/isUndefinedOrNull';
+
+export const useGetSyncStatusMetadata = () => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/hooks/useGetSyncStatusMetadata.ts:5]: this should live in calendar front module (or in modules/settings/accounts) (it's not something generic like object-metadata)
other than than it looks good!

**Review by FelixMalfait — APPROVED** [review]
Thanks @i-am-chitti great contribution as always!

**Pull request merged**